### PR TITLE
Add bits to handle the new codebench layout

### PR DESCRIPTION
**Pull request by kergoth**
If the user hasn't set EXTERNAL_TOOLCHAIN, but CODEBENCH_PATH is set, then we automatically set both EXTERNAL_TOOLCHAIN and EXTERNAL_TARGET_SYS based on the installed toolchains in CODEBENCH_PATH/../toolchains.

### CODE REVIEW

**Comment by kergoth** [comment]
Note the second commit will have an impact on the old layout toolchains as well, it automatically adjusts EXTERNAL_TOOLCHAIN to `<codebench install dir>/codebench` if the user inadvertently set it to the root of the install. I normally don't like such 'magic', but we can easily tell what the user meant, and it displays a warning every time with indications of how they can fix it to silence the warning, which I think is a good compromise, and I hate it when a tool errors and tells me what to do when it could just fix the damn thing for me.

**Comment by kergoth** [comment]
Just a note that doing it this way means it continues to work fine with the existing older layer toolchains as well.

File: classes/codebench.bbclass [hunk 1]
```diff
@@ -41,6 +53,20 @@ python codebench_check () {
     else:
         d.setVar('EXTERNAL_TARGET_SYS', triplets[0])
         d.setVar('EXTERNAL_TOOLCHAIN', os.path.join(toolchains_path, toolchain_subdir))
+
+python codebench_check () {
+    codebench_path = d.getVar('CODEBENCH_PATH', True)
+    exttc = d.getVar('EXTERNAL_TOOLCHAIN', True)
+    if exttc:
+        if codebench_path:
+            # Explicitly set EXTERNAL_TOOLCHAIN is preferred
```
Comment (Noor-Ahsan) [inline classes/codebench.bbclass]: When both paths are set I think we should print a Note that Toolchain Path is preferred and we are picking this patch

Comment (kergoth) [inline classes/codebench.bbclass]: If the EXTERNAL_TOOLCHAIN is invalid, it'll error out, but I can see a warning being useful for the case where both are set and both are valid, so the user is clear on which is being used. Will do.

**Comment by kergoth** [comment]
Warning added, thanks.

**Pull request merged**